File: pyton_files/ex4.py
class Animal:
    def __init__(self, nome):
        self.nome = nome
    def som(self):
        raise NotImplementedError("Metodo som não implementado")

class Cachorro(Animal):
    def som(self):
        return f"{self.nome} Au au!"

class Gato(Animal):
    def som(self):
        return f"{self.nome} Miau!"


meu_cachorro = Cachorro("Dogpoll")
meu_gato = Gato("jiji")
print(meu_cachorro.som())  
print(meu_gato.som())      

File: go_files/Ex4.go
package main

import (
  "fmt"
)

type Animal interface {
  Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
  return "Au au!"
}

type Gato struct{}

func (g Gato) Som() string {
  return "Miau!"
}

func main() {
  var a Animal
  a = Cachorro{}
  fmt.Printf("Cachorro faz: %s\n", a.Som())
  a = Gato{}
  fmt.Printf("Gato faz: %s\n", a.Som())
}